Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -4491,12 +4493,46 @@ public String doCharacterAttemptAttack(ODPAuthenticator auth, CachedEntity user,
         Double charDex = getCharacterDexterity(sourceCharacter);
         Double monsterDex = getCharacterDexterity(targetCharacter);
         Random rnd = new Random();
+        
+        String status = "";
+        
+        Integer autoRunThreshold = (Integer) targetCharacter.getProperty("autoRunThreshold");
+        if(autoRunThreshold == null) autoRunThreshold = 0;
+        Integer autoRunChance = (Integer) targetCharacter.getProperty("autoRunChance");
+        if(autoRunChance == null) autoRunChance = 0;
+        
+        int targetHP = (int) targetCharacter.getProperty("hitpoints");
+        int targetMaxHP = (int) targetCharacter.getProperty("maxHitpoints");
+        
+        //First, we check against the auto run threshold to see if the monster will attempt to run.
+        if((targetHP / targetMaxHP)*100 < autoRunThreshold) {
+            
```

Comment:
`int / int` = `int`. You'll always get a 0 here, unless you cast to double. Do `100d * targetHP / targetMaxHP` instead.